node/tessel-export.js

```diff
@@ -85,6 +94,73 @@ Tessel.prototype.close = function() {
   }, this);
 };
 
+Tessel.prototype.pwmFrequency = function(frequency, cb) {
+  if (frequency < Tessel.pwmMinFrequency || frequency > Tessel.pwmMaxFrequency) {
+    throw new Error(`${frequency} is outside the bounds of ${Tessel.pwmMinFrequency}-${Tessel.pwmMaxFrequency} Hz`);
+  }
+
+  determineDutyCycleAndPrescalar(frequency, (err, results) => {
+    if (err) {
+      throw err;
+    }
+    else {
+      Tessel.pwmPeriod = results.period;
+      Tessel.pwmPrescalar = results.prescalar;
+
+      // We are currently only using TCC Bank 0
+      // This may be expanded in the future to enable PWM on more pins
+      const TCC_ID = 0;
+
+      var packet = new Buffer(4);
+      // Write the command id first
+      packet.writeUInt8(CMD.PWM_PERIOD, 0);
+      // Write our prescalar to the top 4 bits and TCC id to the bottom 4 bits
+      packet.writeUInt8((Tessel.pwmPrescalar << 4) | TCC_ID, 1);
+      // Write our period (16 bits)
+      packet.writeUInt16BE(Tessel.pwmPeriod, 2);
+
+      // Send the packet off to the samd21
+      // on the first port object (regardless of name)
+      this.port[Object.keys(this.port)[0]].sock.write(packet, cb);
```

Comment:
Ooooh fancy. I like the use of `find`. It's one of those functions I can never remember when I'm looking for it.